crates/forge/src/running.rs

```diff
@@ -229,35 +234,42 @@ pub(crate) fn run_test_case(
         &string_to_hint,
     );
 
-    Ok(runner.run_function(
+    match runner.run_function(
         runner.find_function(case.name.as_str())?,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
-    )?)
+    ) {
+        Ok(result) => Ok(TestCaseSummary::from_run_result(result, case, args)),
```

Comment:
Why move the parsing here? It makes whole thing a lot more confusing

Reply:
Will change it, but it requires introducing additional intermediate struct. I need to be able to pass block number with a `RunResult` somehow